internal/mode/static/manager.go

```diff
@@ -268,6 +270,16 @@ func registerControllers(
 				controller.WithK8sPredicate(predicate.ServicePortsChangedPredicate{}),
 			},
 		},
+		{
+			objectType: &apiv1.Service{},
+			options: func() []controller.Option {
+				svcNSName := types.NamespacedName{Namespace: cfg.Namespace, Name: cfg.GatewayPodConfig.ServiceName}
```

Comment:
It's a required field so it'll have to be set. Maybe this could be a configurable field in the CRD in the future, but I didn't consider that option right now.